src/parser/parser.ts

```diff
@@ -200,12 +239,28 @@ export class Parser {
         }
     }
 
+    private static getHotkeyByLine(
+        document: vscode.TextDocument,
+        line: number,
+    ) {
+        const text = CodeUtil.purify(document.lineAt(line).text).trim();
+        // captures hotkeys and hotstrings
+        const hotkey = /^(.+?)::|hotkey(?:\s|,)\s*(?!\s*if)([^,]+)/i.exec(text);
+        if (hotkey) {
+            const hotkeyName = hotkey[1] ?? hotkey[2];
+            return new Hotkey(
+                hotkeyName,
+                document,
+                line,
+                text.indexOf(hotkeyName),
+            );
+        }
+    }
+
     private static getLabelByLine(document: vscode.TextDocument, line: number) {
-        const text = CodeUtil.purify(document.lineAt(line).text);
+        const text = CodeUtil.purify(document.lineAt(line).text).trim();
         // [\u4e00-\u9fa5] Chinese unicode characters
-        const label = /^[ \t]*([\u4e00-\u9fa5_a-zA-Z0-9]+) *:{1}(?!(:|=))/.exec(
-            text,
-        );
+        const label = /^([\u4e00-\u9fa5_a-zA-Z0-9]+):{1}(?!(:|=))/.exec(text);
```

Comment:
Does this regex enforce the "label cannot end with 2 colons" rule? Should it?

Reply:
Yes, it enforce rule. Yes, it should. I revert this changes.

Reply:
Original regex is wrong. ` *:{1}` this part. Spaces not allowed before colon. `Label :` is not label, AutoHotkey shows error.